ui/src/main/java/org/pentaho/di/ui/job/entry/JobEntryDialog.java

```diff
@@ -396,21 +416,36 @@ public EditConnectionListener( CCombo wConnection ) {
 
     @Override public void widgetSelected( SelectionEvent e ) {
       DatabaseMeta databaseMeta = jobMeta.findDatabase( wConnection.getText() );
+      String originalName = databaseMeta.getName();
       if ( databaseMeta != null ) {
         // cloning to avoid spoiling data on cancel or incorrect input
         DatabaseMeta clone = (DatabaseMeta) databaseMeta.clone();
-        String connectionName = showDbDialogUnlessCancelledOrValid( clone, databaseMeta );
-        if ( connectionName != null ) {
+        String editedConnectionName = showDbDialogUnlessCancelledOrValid( clone, databaseMeta );
+        if ( editedConnectionName != null ) {
           // need to replace the old connection with a new one
           try {
-            jobMeta.getDatabaseManagementInterface().removeDatabase( databaseMeta );
-            jobMeta.getDatabaseManagementInterface().addDatabase( clone );
+            DatabaseManagementInterface dbMgr =
+              spoonSupplier.get().getBowl().getManager( DatabaseManagementInterface.class );
+            DatabaseManagementInterface globalDbMgr =
+              DefaultBowl.getInstance().getManager( DatabaseManagementInterface.class );
+            if ( dbMgr.getDatabase( originalName ) != null ) {
+              dbMgr.removeDatabase( databaseMeta );
+              dbMgr.addDatabase( clone );
+            } else if ( globalDbMgr.getDatabase( originalName ) != null ) {
+              globalDbMgr.removeDatabase( databaseMeta );
+              dbMgr.addDatabase( clone );
```

Comment:
this is wrong. Edit should never switch things between levels. The remove and add should always be the same dbMgr.